src/libraries/System.IO.Compression/src/System/IO/Compression/ZipHelper.cs

```diff
@@ -193,5 +196,57 @@ private static bool SeekBackwardsAndRead(Stream stream, byte[] buffer, out int b
                 return true;
             }
         }
+
+        // Converts the specified string into bytes using the optional specified encoding. If null, then the encoding is calculated from the string itself.
+        internal static byte[] EncodeStringToBytes(string? text, Encoding? encoding, out bool isUTF8)
+        {
+            if (text == null)
+            {
+                text = string.Empty;
+            }
+            if (encoding == null)
+            {
+                encoding = GetEncoding(text);
+            }
+            isUTF8 = encoding.CodePage == 65001;
+            return encoding.GetBytes(text);
```

Comment:
```suggestion
            text = text ?? string.Empty;
            encoding = encoding ?? GetEncoding(text);

            isUTF8 = encoding.CodePage == 65001;
            return encoding.GetBytes(text);
```

Reply:
and below

Reply:
I keep forgetting the null coalescing operator already exists. Thanks! 😄

Reply:
> If you don't mind, let's keep this as is. When that bug gets fixed, we can get it changed in bulk via the analyzer.

The suggestion is ready now and it improves code quality and readability, I don't see why we should wait for applying it.

BTW do we include the preamble in the output? If not, why don't we just return `Array.Empty<byte>()` for null and empty strings?

Reply:
I decided to keep the nullable variables as originally intended, so I don't have to use Array.Empty.